docs/design/features/gc-counters-net5.md

```diff
@@ -0,0 +1,43 @@
+# GC Counters for .NET 5
+
+This document describes the new GC EventCounters that are being added in .NET 5.
+
+## Problem
+
+The garbage collector is one of the key runtime components that contribute to the performance characteristics of a .NET application. We added a number of GC counters to CoreCLR in `System.Runtime` provider in .NET Core 3.0. Number of first-party and external customers adopted these counters and have actively been using them in production.
+Currently the exposed GC counters are:
+
+* GC Heap Size
+* % time in GC
+* Gen 0/1/2 Collections Rate
+* Gen 0/1/2 Size
+
+For "deeper" diagnostics on GC performance, one needs to collect a trace via ETW/LTTng/EventPipe and process the trace using TraceEvent programmatically or view it on tools like PerfView/VS. This is not the most convenient way for developers to "easily" see some of the relevant GC metrics, and moreover, the performance overhead of tracing may not be something they can incur in performance-sensitive environments.
+
+In essence, the GC metrics can be thought of as a "basic" performance indicator, as they provide a high-level point-of-view of the GC performance, and can be used to detect regressions in performance tests, or used for monitoring  production environments.
+
+We received feedback from first-party customers (i.e. Exchange, Bing) to increase the number of counters in .NET 5, specifically for the GC, to help them monitor some of their key performance metrics in production. The counters that were requested in the original issues were:
+* GC Heap Fragmentation %
+* Gen 0/1/2 Pause Time
+* Gen 0/1/2 Time Between Each GC
+
+## Solution
+
+The proposal is to add the following GC counters:
+
+* GC Heap Fragmentation %
+* Gen 0/1/2 Pause Time 
```

Comment:
How is the pause time defined for Gen2 GC?

Reply:
It was defined to be the latest Gen 2 GC's pause time... whether that was FGC or BGC. I spoke with Noah a bit and came to the conclusion that it's not worth adding the pause counters (discussed below)